src/sentry/static/sentry/app/utils.jsx

```diff
@@ -23,9 +23,7 @@ export const arrayIsEqual = function(arr, other, deep) {
     return false;
   }
 
-  for (let i = 0, l = Math.max(arr.length, other.length); i < l; i++) {
-    return valueIsEqual(arr[i], other[i], deep);
-  }
+  return arr.every((val, idx) => valueIsEqual(val, other[idx], deep));
```

Comment:
this will eliminate some corner case false positive matches. afaik it's not used too extensively in the app